m4opt/missions/_ultrasat/__init__.py

```diff
@@ -20,11 +20,24 @@
 from . import data
 
 
-def _read_skygrid():
-    table = Table.read(
-        resources.files(data) / "LCS_nonoverlapping_grid.csv", format="ascii.csv"
-    )
-    return SkyCoord(table["RA"], table["Dec"], unit=u.deg)
+def _read_skygrid(allsky=True) -> SkyCoord:
+    # Load the non-overlapping LCS grid or the allsky grid.
+    # By default, the allsky grid is loaded.
+    if allsky:
+        table = Table.read(
+            resources.files(data) / "AllSS_grid_361.txt", format="ascii.no_header"
+        )
+        ra = [float(str(s).strip().rstrip(",")) for s in table["col1"]]
+        dec = table["col2"]
+
+    else:
+        table = Table.read(
+            resources.files(data) / "LCS_nonoverlapping_grid.csv", format="ascii.csv"
+        )
+        ra = table["RA"]
+        dec = table["Dec"]
```

Comment:
These two branches are so dissimilar that it would be better as two separate functions, rather than one function with an argument.